packages/graphile-build-pg/src/plugins/PgTypesPlugin.js

```diff
@@ -994,30 +998,58 @@ end`;
           type.type === "d" &&
           type.domainBaseTypeId
         ) {
-          const baseType = getGqlTypeByTypeIdAndModifier(
-            type.domainBaseTypeId,
-            typeModifier
-          );
-          const baseInputType = getGqlInputTypeByTypeIdAndModifier(
-            type.domainBaseTypeId,
-            typeModifier
-          );
-          // Hack stolen from: https://github.com/graphile/postgraphile/blob/ade728ed8f8e3ecdc5fdad7d770c67aa573578eb/src/graphql/schema/type/aliasGqlType.ts#L16
-          gqlTypeByTypeIdAndModifier[type.id][typeModifierKey] = Object.assign(
-            Object.create(baseType),
-            {
-              name: inflection.domainType(type),
-              description: type.description,
-            }
-          );
-          if (baseInputType && baseInputType !== baseType) {
+          // might be used as an enum alias: #1500
+          if (
+            type.name.endsWith("_enum_domain") ||
+            (type.tags && type.tags.enum)
+          ) {
+            const underlyingEnumTypeName = type.name.endsWith("_enum_domain")
+              ? type.name.replace("_enum_domain", "")
+              : type.tags.enum;
```

Comment:
Tag is more explicit, so it should "win".

`.replace` isn't safe in case `_enum_domain` occurs more than once.

```suggestion
          const tagEnumName =
            type.tags && typeof type.tags.enum === "string" ? type.tags : null;
          // TODO: move this to the top of the file
          const ENUM_DOMAIN_SUFFIX = "_enum_domain";
          const truncatedTypeName =
            !tagEnumName && type.name.endsWith(ENUM_DOMAIN_SUFFIX)
              ? type.name.slice(0, type.name.length - ENUM_DOMAIN_SUFFIX.length)
              : null;
          const underlyingEnumTypeName = tagEnumName || truncatedTypeName;
          if (underlyingEnumTypeName !== null) {
```